nameko/messaging.py

```diff
@@ -98,6 +106,7 @@ def spam(self, data):
         """
         self.exchange = exchange
         self.queue = queue
+        self.serializer = serializer
```

Comment:
when `setup` is called this value is ignored, and indeed your docstring states it will always be overriden. So why maintain the value on the instance at all? is there a case when `setup` is not called and we determine the correct serializer from this value?